src/backend/api/Fusion.Resources.Domain/Commands/SecondOpinions/GetSecondOpinions.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            RequestId = requestId;
+            return this;
+        }
+
+        public GetSecondOpinions WithAssignee(PersonId assigneeId)
+        {
+            AssigneeId = assigneeId;
+            return this;
+        }
+
+
+        public GetSecondOpinions WithId(Guid secondOpinionId)
+        {
+            Id = secondOpinionId;
+            return this;
+        }
+
+
+        public GetSecondOpinions WithQuery(ODataQueryParams query)
+        {
+            Query = query;
+            return this;
+        }
+
+        public class Handler : IRequestHandler<GetSecondOpinions, List<QuerySecondOpinion>>
+        {
+            private readonly ResourcesDbContext db;
+            private readonly IProfileService profileService;
+
+            public Handler(ResourcesDbContext db, IProfileService profileService)
+            {
+                this.db = db;
+                this.profileService = profileService;
+            }
+
+            public async Task<List<QuerySecondOpinion>> Handle(GetSecondOpinions request, CancellationToken cancellationToken)
+            {
+                var creatorId = default(Guid?);
+                var assigneeId = default(Guid?);
+
+                if (request.CreatedById.HasValue)
+                {
+                    var creator = await profileService.EnsurePersonAsync(request.CreatedById.Value);
+                    creatorId = creator?.Id;
+                }
+
+                if (request.AssigneeId.HasValue)
+                {
+                    var assignee = await profileService.EnsurePersonAsync(request.AssigneeId.Value);
+                    assigneeId = assignee?.Id;
+                }
+
+                var query = db.SecondOpinions
+                    .Include(x => x.Responses.Where(r => r.AssignedToId == assigneeId || assigneeId == null || r.State == DbSecondOpinionResponseStates.Published))
+                        .ThenInclude(x => x.AssignedTo)
+                    .Include(x => x.CreatedBy)
+                    .Include(x => x.Request).ThenInclude(x => x.Project)
+                    .Include(x => x.Request).ThenInclude(x => x.CreatedBy)
+                    .AsQueryable();
+
+                if (request.Id.HasValue)
+                {
+                    query = query.Where(r => r.Id == request.Id);
+                }
+
+                if (creatorId.HasValue)
+                {
+                    query = query.Where(x => x.CreatedById == creatorId);
+                }
+
+                if (assigneeId.HasValue)
+                { 
+                    query = query.Where(x => x.Responses.Any(x => x.AssignedToId == assigneeId));
+                }
+
+                if (request.RequestId.HasValue)
+                {
+                    query = query.Where(x => x.RequestId == request.RequestId);
+                }
+
+                if(request.Query.HasFilter)
+                {
+                    var filter = request.Query.Filter.GetFilterForField("state");
```

Comment:
Can't you just filter on what you pass along ?
Also. Have you seen the extension => query.ApplyOdataFilters(xxxxxxx ?

Reply:
Ser fremdeles ikke helt noen god grunn til å endre på filter input